assets/js/payment-methods-demo/stripe-utils/utils.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Internal dependencies
+ */
+import { normalizeLineItems } from '../express-payment/apple-pay/normalize';
+import { errorTypes, errorCodes } from '../payment-methods/stripe';
+
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('./type-defs').StripeServerData} StripeServerData
+ * @typedef {import('./type-defs').StripePaymentItem} StripePaymentItem
+ * @typedef {import('./type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/cart').CartTotalItem} CartTotalItem
+ */
+
+// @todo this can't be in the file because the block might not show up unless it's
+// rendered! (which also means it won't get passed from the server until it is rendered).
+// so we need to work out a loading process for when the cart or checkout hasn't been added to the
+// content yet - definitely don't want to load stripe without the block in the page. So that means
+// checkout will need to have some sort of editor preview for stripe.
+/**
+ * Stripe data comes form the server passed on a global object.
+ *
+ * @return  {StripeServerData}
+ */
+const getStripeServerData = () => {
```

Comment:
So if I understand correctly, gateways might need to pass data to themselves via settings,  but we need to ensure block checkout is loaded first?

How about a common API on the PHP side for gateways to insert variables, and then have either the checkout or payment provider load them in somehow and pass through to payment method on initialisation.

Reply:
I like that idea. Adding as a follow-up.